src/indra_cogex/sources/pubmed/__init__.py

```diff
@@ -0,0 +1,92 @@
+import csv
+import json
+import logging
+import pystow
+from indra_cogex.representation import Node, Relation
+from indra_cogex.sources.processor import Processor
+
+
+logger = logging.getLogger(__name__)
+
+resources = pystow.module("indra", "cogex", "pubmed")
+
+MESH_PMID = resources.join(name="mesh_pmids.csv")
+PMID_YEAR = resources.join(name="pmid_years_07-2021.json")
+TEXT_REFS = resources.join(name="text_refs.tsv")
+
+
+class PubmedProcessor(Processor):
+    name = "pubmed"
+
+    def __init__(
+        self,
+        mesh_pmid_path=MESH_PMID,
+        pmid_year_path=PMID_YEAR,
+        text_refs_path=TEXT_REFS,
+    ):
+        self.mesh_pmid_path = mesh_pmid_path
+        self.pmid_year_path = pmid_year_path
+        self.text_refs_path = text_refs_path
+
+    def get_nodes(self):
+        # We iterate over text refs to get the nodes and
+        # then look up the year to add as a property
+        pmid_node_type = "Publication"
+        logger.info("Loading PMID year info from %s" % self.pmid_year_path)
+        with open(self.pmid_year_path, "r") as fh:
+            pmid_years = json.load(fh)
+        logger.info("Loaded PMID year info from %s" % self.pmid_year_path)
+
+        def get_val(val):
+            # postgres exports \N for missing values
+            if val == "\\N":
+                return None
+            else:
+                return val
+
+        with open(self.text_refs_path, "r") as fh:
+            reader = csv.reader(fh, delimiter="\t")
+            for trid, pmid, pmcid, doi, pii, url, manuscript_id in reader:
+                if not get_val(pmid):
+                    continue
+                year = pmid_years.get(pmid, None)
+                data = {
+                    "trid": get_val(trid),
+                    "pmcid": get_val(pmcid),
+                    "doi": get_val(doi),
+                    "pii": get_val(pii),
+                    "url": get_val(url),
+                    "manuscript_id": get_val(manuscript_id),
+                    "year": year,
+                }
+                yield Node("PUBMED", pmid, labels=[pmid_node_type], data=data)
```

Comment:
maybe better to use `"ARTICLE"` in case there are ever other resources that don't have pubmed identifiers

Reply:
No but this is specifically a prefix for a PubMed ID so it has to be PUBMED.